### ISSUE TITLE
Replying to deleted comments is prevented inconsistently

### ISSUE THREAD

**Issue by kaberett** [dreamwidth/dw-free#1591]
If you click on the "Reply to this comment" link in an e-mail notification about a comment that's subsequently been deleted, you get a very basic page the full text of which is "This comment has been deleted; you cannot reply to it."

However, if you reply to the deleted comment by replying to the e-mail, your comment is posted with the parent "(deleted comment)". <a href="http://kaberett.dreamwidth.org/456861.html?thread=4891805#cmt4891805">Example</a>.

Behaviour should be consistent. (Perhaps by bouncing back an error message if you reply-by-email to a deleted comment?)

**Comment by rahaeli** [comment]
Yeah, a bounce-back error message (with your comment text!) is probably best.

**Comment by kaberett** [comment]
It occurs to me that if I get around to doing this bug I should double-check that replying to comments on posts you can't see/in journals you've been banned from by e-mail exhibits the correct behaviour.

**Comment by kaberett** [comment]
... claim.

**Comment by kaberett** [comment]
... oh. If you click through to ?mode=reply on a comment and that comment then gets deleted, you can <i>also</i> successfully post a reply from the page you kept open. I tested this after poking around at LJ::Protocol revealed no error code for "the thing you were trying to reply to has been deleted" -- it seems to  me that this is something where I might well want to introduce an error code and then check *at the point a reply is being posted* in the general case, rather than just in the instance of reply-by-email? (@kareila?)

**Comment by kaberett** [comment]
@rahaeli -- could I get some input on what you'd ideally like to have happen here? <3

**Comment by rahaeli** [comment]
Sorry! Yeah, if a comment is deleted, it should be unreplyable-to by any means. I have no opinion as to whether it should be a new protocol method or whether it should be an outside-protocol check in whatever function that checks if you can reply to a comment, but it's probably better to go for whichever's easiest.

**Comment by kareila** [comment]
Probably best to do the check in protocol, since the entire point of the protocol as I understand it is to make behavior of various actions consistent regardless of whether they were initiated via website, email, or API.

**Comment by kareila** [comment]
... And yeah, "We checked the entry/comment/user status when we opened the form/sent the email but not when we received the submission" is a pretty common failure mode, sadly.
